build/source/engine/t2enthalpy.f90

```diff
@@ -0,0 +1,468 @@
+    case(iname_soil);    iLayer = ixControlIndex + nSnow
+    case(iname_aquifer); cycle ! aquifer: do nothing (no thermodynamics in the aquifer)
+    case default; err=20; message=trim(message)//'expect case to be iname_cas, iname_veg, iname_snow, iname_soil, iname_aquifer'; return
+   end select
+
+   ! identify domain
+   select case(ixDomainType)
+
+    ! -----
+    ! - canopy air space...
+    ! ---------------------
+    case(iname_cas)
+     err=20; message=trim(message)//'still need to implement canopy air space'
+     return
+
+    ! -----
+    ! - vegetation canopy...
+    ! -----------------------
+    case(iname_veg)
+     err=20; message=trim(message)//'still need to implement vegetation canopy'
+     return
+
+    ! -----
+    ! - snow...
+    ! ---------
+    case(iname_snow)
+     err=20; message=trim(message)//'still need to implement snow'
+     return
+
+    ! -----
+    ! - soil...
+    ! ---------
+    case(iname_soil)
+
+     ! make association to variables in the data structures...
+     soilVars: associate(&
+
+     ! associate model parameters
+     soil_dens_intr => mpar_data%var(iLookPARAM%soil_dens_intr)%dat(ixControlIndex)      , & ! intrinsic soil density             (kg m-3)
+     theta_sat      => mpar_data%var(iLookPARAM%theta_sat)%dat(ixControlIndex)           , & ! soil porosity                      (-)
+     theta_res      => mpar_data%var(iLookPARAM%theta_res)%dat(ixControlIndex)           , & ! volumetric residual water content  (-)
+     vGn_alpha      => mpar_data%var(iLookPARAM%vGn_alpha)%dat(ixControlIndex)           , & ! van Genuchten "alpha" parameter    (m-1)
+     vGn_n          => mpar_data%var(iLookPARAM%vGn_n)%dat(ixControlIndex)               , & ! van Genuchten "n" parameter        (-)
+
+     ! associate values in the lookup table
+     Tk            => lookup_data%z(ixControlIndex)%var(iLookLOOKUP%temperature)%lookup  , & ! temperature (K)
+     Ey            => lookup_data%z(ixControlIndex)%var(iLookLOOKUP%enthalpy)%lookup     , & ! enthalpy (J m-3)
+     E2            => lookup_data%z(ixControlIndex)%var(iLookLOOKUP%deriv2)%lookup         & ! second derivative of the interpolating function
+
+     ) ! end associate statement
+
+     ! diagnostic variables
+     vGn_m    = 1._dp - 1._dp/vGn_n
+     Tcrit    = crit_soilT( mLayerMatricHeadTrial(ixControlIndex) )
+     vFracWat = volFracLiq(mLayerMatricHeadTrial(ixControlIndex),vGn_alpha,theta_res,theta_sat,vGn_n,vGn_m)
+
+     ! *** compute enthalpy of water for unfrozen conditions
+     if(mlayerTempTrial(iLayer) > Tcrit)then
+      enthWater = iden_water*Cp_water*vFracWat*(mlayerTempTrial(iLayer) - Tfreeze) ! valid for temperatures below freezing also
+      enthPhase = 0._dp
+
+     ! *** compute enthalpy of water for frozen conditions
+     else
+
+      ! calculate enthalpy at the temperature (cubic spline interpolation)
+      call splint(Tk,Ey,E2,mlayerTempTrial(iLayer),enthTemp,err,cmessage)
```

Comment:
These tables are calculated assuming saturated soil. Does that assumption hold when we get to this point in the code?